src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,228 @@
+    merge_df = base_df.merge(intersecting_df, on=join_on, how='left', indicator=True)
+    filtered_df = merge_df[merge_df['_merge'] == 'left_only'].drop(columns=['_merge']).rename(
+        columns=lambda x: x[:-2] if x.endswith('_x') else x)[list(base_df.columns)]
+    return filtered_df
+
+
+# TODO
+def sync_synonyms(
+    ontology_db_path: Union[Path, str], mondo_synonyms_path: Union[Path, str],
+    mondo_mappings_path: Union[Path, str], onto_config_path: Union[Path, str], outpath_added: Union[Path, str],
+    outpath_confirmed: Union[Path, str], outpath_deleted: Union[Path, str], outpath_updated: Union[Path, str]
+):
+    """Create outputs for syncing synonyms between Mondo and its sources."""
+    # Iterate over mappings
+    mappings_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#')
+    # - filter exact matches
+    mappings_df = mappings_df[mappings_df['predicate_id'] == 'skos:exactMatch']
+    # - rename cols
+    mappings_df = mappings_df.rename(columns={'subject_id': 'mondo_id', 'subject_label': 'mondo_label', 'object_id': 'source_id',
+        'object_label': 'source_label'})
+    # - remove unneeded cols
+    mappings_df = mappings_df[['mondo_id', 'mondo_label', 'source_id', 'source_label']]
+    # todo: refactor filtering by prefixes
+    #  check if this pattern has been used elsewhere; I think it has. and abstract into func for these 3 lines; 'filter prefixes'
+    #  - OR use this?: get_all_owned_terms()
+    #    - it seems long and messy compared to these 3 lines, but is more comprehensive. I could make get_all_owned_terms() a simple 3 line func and then make a separate func for all edge cases?
+    # - filter by source
+    owned_prefix_map: PREFIX_MAP = get_owned_prefix_map(onto_config_path)
+    mappings_df['source_prefix'] = mappings_df['source_id'].apply(lambda x: x.split(':')[0])
+    mappings_df = mappings_df[mappings_df['source_prefix'].isin(owned_prefix_map.keys())]
+    del mappings_df['source_prefix']
+    # - filter obsoletes (from mapping set --> source & mondo)
+    #  - filter obsoleted in Mondo
+    mappings_df = mappings_df[~mappings_df['mondo_label'].str.startswith('obsolete')]
+    #  - filter obsoleted in source
+    source_db: SqlImplementation = get_adapter(ontology_db_path)
+    source_obsoletes: Set[CURIE] = set([x for x in source_db.obsoletes()])
+    mappings_df = mappings_df[~mappings_df['source_id'].isin(source_obsoletes)]
+    # - filter mapped but deleted IDs
+    #   alternative ways: entities(). all_entity_curies(). node() (empty props), node_exists() (falsely says 'True',
+    #   prolly cuz erroneous xrefs exist)
+    source_labels: List[Tuple[CURIE, str]] = [x for x in source_db.labels(mappings_df['source_id'].tolist())]
+    source_deleted_ids: Set[CURIE] = set([x[0] for x in source_labels if not x[1]])
+    mappings_df = mappings_df[~mappings_df['source_id'].isin(source_deleted_ids)]
+    # - update source labels (sometimes Mondo & source have discrepancy)
```

Comment:
## FYI: Update outdated labels
Sometimes the labels for mapped terms in `mondo.sssom.tsv` are incorrect. In the case of OMIM, 1,740 of 10,114 (~17%). So I update the labels here to pull from the source.

Reply:
Interesting find! If the external source label is incorrect in the `mondo.sssom.tsv` file I gather it's fine to use the label directly from the source. 

For the numbers you included, are you counting OMIM entries in `mondo.sssom.tsv` file? I see 10,398 total entries in the `mondo.sssom.tsv` starting with "omim".

Do you have a few examples that can be added to a new issue to fix what's happening here with the label being incorrect in the `mondo.sssom.tsv` file?

Reply:
Yes; it must have slightly fewer lines due to the filtering of obsoletes.

Issue created!:
- https://github.com/monarch-initiative/mondo/issues/7968

Marking this resolved.